lib/block-supports/layout.php

```diff
@@ -838,6 +846,25 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		$block_gap             = $global_settings['spacing']['blockGap'] ?? null;
 		$has_block_gap_support = isset( $block_gap );
 
+		/*
+		 * We generate a unique ID based on all the data required to obtain the
+		 * corresponding layout style. This way, the CSS class names keep the same
+		 * even for different blocks with the same layout definition. We need this to
+		 * make the CSS class names stable across paginations for features like the
+		 * enhanced pagination of the Query block.
+		 */
+		$container_class = gutenberg_unique_prefixed_id_from_array(
+			array(
+				$used_layout,
+				$has_block_gap_support,
+				$gap_value,
+				$should_skip_gap_serialization,
+				$fallback_gap_value,
+				$block_spacing,
+			),
+			'wp-container-' . sanitize_title( $block['blockName'] ) . '-is-layout-'
+		);
+
```

Comment:
Looking at the implementation, could the entire code be simplified to:

```php
		// Only add container class and enqueue block support styles if unique styles were generated.
		if ( ! empty( $style ) ) {
			$class_names[] = 'wp-container-' . sanitize_title( $block['blockName'] ) . '-is-layout-' . substr( md5( $style ), 0, 8 );
		}
```

In theory, that approach might also help to avoid running the logic when no class is needed.

Reply:
OK, I missed that you must generate the class name before calling `gutenberg_get_layout_style`. That's a limitation as it prevents writing more optimal code. However, using a util to generate a hash based on the params should be good, too.

Reply:
Yup, exactly. That's why I decided to compute the hash using all the parameters required to generate `$style` rather than `$style` itself.